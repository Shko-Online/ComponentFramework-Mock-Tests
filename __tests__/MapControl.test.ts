/*
	Unless explicitly acquired and licensed from Licensor under another
	license, the contents of this file are subject to the Reciprocal Public
	License ("RPL") Version 1.5, or subsequent versions as allowed by the RPL,
	and You may not copy or use this file in either source code or executable
	form, except in compliance with the terms and conditions of the RPL.

	All software distributed under the RPL is provided strictly on an "AS
	IS" basis, WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, AND
	LICENSOR HEREBY DISCLAIMS ALL SUCH WARRANTIES, INCLUDING WITHOUT
	LIMITATION, ANY WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
	PURPOSE, QUIET ENJOYMENT, OR NON-INFRINGEMENT. See the RPL for specific
	language governing rights and limitations under the RPL. 
*/

import * as sinon from "sinon";

import { ComponentFrameworkMockGenerator } from "@shko-online/componentframework-mock/ComponentFramework-Mock-Generator";
import {
  IInputs,
  IOutputs,
} from "@powerapps-samples/map-control/MapControl/generated/ManifestTypes";
import { MapControl } from "@powerapps-samples/map-control/MapControl";
import { StringPropertyMock } from "@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/StringProperty.mock";


describe("MapControl", () => {
  let mockGenerator: ComponentFrameworkMockGenerator<IInputs, IOutputs>;
  beforeEach(() => {
    const container = document.createElement('div');

  mockGenerator = new ComponentFrameworkMockGenerator(
    MapControl,
    {
    controlApiKey: StringPropertyMock,
    controlValue: StringPropertyMock
  }, container)
});
  afterEach(() => {
    document.body.innerHTML = null;
  });
  it("Init should work", () => {
    mockGenerator.ExecuteInit();
    sinon.assert.calledOnce(mockGenerator.control.init);
    expect(document.body).toMatchSnapshot();
  });
  it("Update View should Work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });

  it("MapUrl not null", () => {
    (mockGenerator.context.parameters.controlValue as StringPropertyMock).setValue("MapUrl");
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });
  it("ApiKey not null", () => {
    (mockGenerator.context.parameters.controlValue as StringPropertyMock).setValue("MapUrl");
    (mockGenerator.context.parameters.controlApiKey as StringPropertyMock).setValue("ApiKey");
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });
});

