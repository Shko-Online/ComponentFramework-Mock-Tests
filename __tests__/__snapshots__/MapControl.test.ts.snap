// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapControl ApiKey not null 1`] = `<body />`;

exports[`MapControl Init should work 1`] = `<body />`;

exports[`MapControl Update View should Work 1`] = `<body />`;

exports[`MapControl Update View should Work 2`] = `<body />`;

exports[`MapControl Update View should Work 3`] = `<body />`;
